cargo-near/src/util/step_printer.rs

```diff
@@ -0,0 +1,41 @@
+use colored::Colorize;
+
+pub struct StepPrinter {
+    cur_step: u32,
+    max_steps: u32,
+}
+
+impl StepPrinter {
+    pub fn new(max_steps: u32) -> Self {
+        Self {
+            cur_step: 0,
+            max_steps,
+        }
+    }
+
+    pub fn print_step(&mut self, msg: &str) {
+        self.cur_step += 1;
+        if self.cur_step > self.max_steps {
+            panic!(
+                "Expected process to have exactly {} steps, but encountered an extra step #{}",
+                &self.max_steps, &self.cur_step
+            )
+        }
+        eprintln!(
+            " {} {}",
+            format!("[{}/{}]", &self.cur_step, &self.max_steps).bold(),
+            msg.green().bold()
```

Comment:
This is a little too strict, don't you think? ideally, we should be able to step print any colors at the caller site.

Reply:
Well, the idea is to print uniform build step messages. Kind of like `cargo` prints `Updating`/`Compiling`/`Finished` etc in the same boldness and color. Do you think there is an argument for this to be customizable on the caller site?

Reply:
Was just thinking of something like `Skipped embedding` which seems more like a filler than an actual step. Making it green seems like it accomplishes something, whereas it doesn't.